### PR TITLE
Modal Integration

### PR DESCRIPTION
**Pull request by marcusfilipesr**
This PR is devoted to implementing a new integration inside ROSS, using the modal reduction. The main goal is to enable the solution of #429, #530 and #884.

This PR itself, does not solve any of the cited issues yet, i.e., it is a pre solution.

- [x] Insert the modal analysis to perform the time integration.
- [ ] Rearrange the integrate_solver file.
- [ ] Verify robustness of the automatic number of modes. ( Forced inputs? )
- [ ] Run benchmark code with different number of modes evaluating the error

### CODE REVIEW

**Comment by marcusfilipesr** [comment]
I've just implementend the integration method. It's lacking the tests, which I'll be implementing tomorrow.

**Comment by marcusfilipesr** [comment]
@raphaeltimbo I think I've finished the first step which was:
-  Insert the modal analysis to perform the time integration.

I think it would be better to go slowly, so I will keep this branch without changes until your review. What do you think?

File: ross/rotor_assembly.py [hunk 1]
```diff
@@ -1350,6 +1350,215 @@ def _lti(self, speed, frequency=None):
...
+        ModMat = ModMat
+
+        # Modal transformations
+        self.Mmodal = ((ModMat.T).dot(M)).dot(ModMat)
+        self.Cmodal = ((ModMat.T).dot(C)).dot(ModMat)
+        self.Gmodal = ((ModMat.T).dot(G)).dot(ModMat)
+        self.Kmodal = ((ModMat.T).dot(K)).dot(ModMat)
+        self.Kstmodal = ((ModMat.T).dot(Kst)).dot(ModMat)
+
+        self.Fmodal = (ModMat.T).dot(F.T)
```
Comment (raphaeltimbo) [inline ross/rotor_assembly.py:1497]: I think it is easier to ready the code if we use the mat_mul operator (@) for matrix multiplication.

**Review by raphaeltimbo — CHANGES_REQUESTED** [review]
Made some small comments for now. 
I think we should, in parallel, solve the issue #978, so that we can benchmark this new integrator.

File: ross/rotor_assembly.py [hunk 2]
```diff
@@ -1365,9 +1365,13 @@ def _modal_analysis(self, speed, F, t, number_modes=12, method="rk45"):
             Each column corresponds to a dof and each row to a time.
         t : array
             Time array.
-        number_modes : int
-            Number of modes considered in the modal reduction. Default is 12.
-        method : str
+        number_modes : str, int, optional
+            Number of modes considered in the modal reduction. It is possible to pass the following string:
+
+                'force' : the code will set the number of modes based on the rotor natural frequencies.
```
[the remark this answers is not in the document]
Comment (marcusfilipesr) [reply ross/rotor_assembly.py:1371]: I'll change it right away.

File: ross/rotor_assembly.py (hunk 1 above)
Comment (marcusfilipesr) [reply ross/rotor_assembly.py:1497]: Ok

**Comment by ross-bott** [comment]
Hi there!
I have marked this issue as stale because it has not had activity for 45 days.
Consider the following options:
- If the issue refers to a large task, break it in smaller issues that can be solved in
less than 45 days;
- Label the issue as `wontfix` or `wontfix for now` and close it.

**Comment by raphaeltimbo** [comment]
We have implemented the newmark integration.

**Pull request closed without merging**